test/jdk/java/awt/im/4959409/bug4959409.java

```diff
@@ -22,51 +22,179 @@
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+public class bug4959409 {
+
+    public final static int TIMEOUT = 30;
+    public final static int DELAY = 300;
+    private static JFrame frame;
+    private static JTextField jTextField;
+    private static JLabel jLabel;
+
+    public JFrame createUIAndTest() throws Exception {
+        final CountDownLatch frameVisibleLatch = new CountDownLatch(1);
+        final CountDownLatch componentVisibleLatch = new CountDownLatch(1);
+        final CountDownLatch keyPressedEventLatch = new CountDownLatch(1);
+        final Point points[] = new Point[1];
+        final Rectangle rect[] = new Rectangle[1];
+
+        SwingUtilities.invokeAndWait(() -> {
+            frame = new JFrame("Test bug4959409");
+            jTextField = new JTextField();
+            jLabel = new JLabel();
+            frame.setLayout(new BorderLayout());
+            frame.addComponentListener(new ComponentAdapter() {
+                @Override
+                public void componentShown(ComponentEvent e) {
+                    super.componentShown(e);
+                    System.out.println("Frame is visible " + e.toString());
+                    frameVisibleLatch.countDown();
+                }
+            });
+
+            jTextField.addComponentListener(new ComponentAdapter() {
+                @Override
+                public void componentShown(ComponentEvent e) {
+                    super.componentShown(e);
+                    System.out.println("Component is visible + " + e.toString());
+                    componentVisibleLatch.countDown();
+                }
+            });
 
-public class bug4959409 extends javax.swing.JApplet {
-    public void init() {
-        new TestFrame();
+            jTextField.addKeyListener(new KeyAdapter() {
+
+                @Override
+                public void keyPressed(KeyEvent keyEvent) {
+                    super.keyPressed(keyEvent);
+                    int code = keyEvent.getKeyCode();
+                    int mod = keyEvent.getModifiersEx();
+                    if (code == '1' && mod == KeyEvent.SHIFT_DOWN_MASK) {
+                        keyPressedEventLatch.countDown();
+                        jLabel.setText("KEYPRESS received for Shift+1");
+                        System.out.println("KEYPRESS received for Shift+1");
+                    } else {
+                        jLabel.setText("Did not received KEY PRESS for Shift+1");
+                        System.out.println("Did not received KEY PRESS for Shift+1");
+                    }
+                }
+            });
+
+            Container container = frame.getContentPane();
+            container.add(jTextField, BorderLayout.SOUTH);
+            container.add(jLabel, BorderLayout.CENTER);
+            frame.setSize(300, 300);
+            frame.setAlwaysOnTop(true);
+            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+            frame.setVisible(true);
```

Comment:
A small suggestion is to move the frame to the center of the screen via frame.setLocationRelativeTo(null); before "frame.setVisible(true);"

Reply:
Added frame.setLocationRelativeTo(null); & tested now I can see the frame is shown at the center of the screen.